src/components/TaskDetail.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from 'react';
+import MDEditor from '@uiw/react-md-editor';
+import { useAuth } from '../contexts/AuthContext';
+import { useTheme } from '../contexts/ThemeContext';
+import rehypeSanitize from 'rehype-sanitize';
+
+export function TaskDetail({ task, isOpen, onClose, onUpdate, onDelete }) {
+  const { user } = useAuth();
+  const { isDark } = useTheme();
+  const [description, setDescription] = useState(task?.description || '');
+  const [isEditing, setIsEditing] = useState(false);
+  const [showDeleteConfirm, setShowDeleteConfirm] = useState(false);
+
+  useEffect(() => {
+    if (task) {
+      setDescription(task.description || '');
+    }
+  }, [task]);
+
+  if (!isOpen || !task) return null;
+
+  const handleSave = () => {
+    onUpdate(task.id, { description });
+    setIsEditing(false);
+  };
+
+  const handleDelete = () => {
+    if (showDeleteConfirm) {
+      onDelete(task.id);
+      onClose();
+    } else {
+      setShowDeleteConfirm(true);
+    }
+  };
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const handleDelete = async () => {
  try {
    if (showDeleteConfirm) {
      await onDelete(task.id);
      onClose();
    } else {
      setShowDeleteConfirm(true);
    }
  } catch (error) {
    console.error('Failed to delete task:', error);
    // Handle error appropriately
  }
};
```
The handleDelete function should be wrapped in a try-catch block to handle potential errors during deletion. This ensures graceful error handling if the delete operation fails.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;